pages/wiki/openembedded.md

```diff
@@ -3,10 +3,243 @@ title: OpenEmbedded
+
+```
+DEPENDS += "qml-asteroid asteroid-generate-desktop-native qttools-native qtdeclarative-native"
+```
+
+With those two small changes, we should now get a successful build.
+
+As before, we can easily test on the watch:
+
+```
+devtool deploy-target hello root@192.168.2.15
+```
+
+This will automatically install all of the install targets from the `hello` project onto the watch. Start the launcher on the watch and scroll to the end of your list of apps, and you should now see a "HELLO WORLD APP" at the end. Running it will show "Hello World!" on the screen.
+
+Removing it from the watch is also the same as before:
+
+```
+devtool undeploy-target hello root@192.168.2.15
+```
+
+
+# Sharing your changes
+
+If you make a change to AsteroidOS software that you think might be useful to others, such as a bug fix or feature enhancement, there are several ways to share it, depending on what exactly is being changed.
+
+ - creating a recipe patch file
+ - creating a pull request for an AsteroidOS project
+ - creating an installable package (`.ipk`)
+
+## Creating a recipe patch file
+
+Many of the software packages that make up an AsteroidOS image are not unique to AsteroidOS. They may be general purpose tools such as `systemctl` or watch-specific software from other open source repositories such as `mcetool`. For these, there may be a need to introduce a patch just for AsteroidOS. To do so, first, within the appropriate `workspace` directory where you made changes, use `git commit` to commit the changes to the local repository. Once this is done, automatically create the patch. For our example, we could do that with this command:
+
+```
+devtool finish asteroid-flashlight meta-asteroid
+```
+
+This creates one patch for each new git commit and automatically updates the recipe file in the named layer (in this case, `meta-asteroid`). (Note that this is just an example and would not really be appropriate for `asteroid-flashlight` because this application is unique to AsteroidOS.)  To see what was done, you can navigate to `asteroid/src/meta-asteroid` and type `git diff`. You will see that the patch file has been added to the recipe for `asteroid-flashlight`. Also note that to ensure you don't lose any work, `devtool` will not delete the directory tree in `workspace` unless you specifically ask it to do so.
+
+## Create a pull request
+
+For an AsteroidOS application, it may be more appropriate to create a [*pull request*](https://www.atlassian.com/blog/git/written-unwritten-guide-pull-requests). To do that, we need to create a fork of the software we want to modify, create a branch, make the changes, commit and push the changes, and then create a pull request (PR). This guide doesn't describe all of these steps in detail, but describes how to use `devtool` and `git` to create a PR.
```

Comment:
Maybe worth linking to something like https://opensource.com/article/19/7/create-pull-request-github ? Instead of saying "Create a pull request" "We won't tell you how to send a pull request" ?

Reply:
That's a much better link.  Thanks!